src/analysis/functions.rs

```diff
@@ -130,15 +130,14 @@ pub fn analyze<F: Borrow<library::Function>>(
     'func: for func in functions {
         let func = func.borrow();
         let configured_functions = obj.functions.matched(&func.name);
-        let mut status = GStatus::Generate;
+        let mut status = obj.status;
         for f in configured_functions.iter() {
             match f.status {
                 GStatus::Ignore => continue 'func,
-                GStatus::Manual => {
-                    status = GStatus::Manual;
+                s => {
+                    status = s;
                     break;
                 }
-                GStatus::Generate => (),
             }
         }
```

Comment:
I'm not really sure what to do here. It is very counter-intuitive to have multiple configurations specifying a different mode. IMO we should assert that all statuses are equal. OTOH that might make it weird when using patterns to apply generic configuration, then later `manual/ignore` a few specific cases?

This now breaks or continues on the first item anyway, and clippy will tell us that this doesn't loop.

Because of using the first item (instead of the first encounter of `ignore = true`) this actually uncovers a little inconsistency in GStreamer's `Element` where the `"seek"` and `"seek_simple"` functions first receive a `bool_return_is_error`, then later down are reconfigured for `manual = true`.

Reply:
The best course of action is probably to revert this change as it's not necessary to fix the issue described in the first commit, though semantics change slightly if a parent object would have been set to manual, but a function set to generate (which is not something we support now, but maybe something to recognize in the future).

In addition, the config parser should check and warn/fail on duplicate `Ident::Name`.